PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.API/Middleware/ValidationMiddleware.cs

```diff
@@ -0,0 +1,66 @@
+using FluentValidation;
+using FluentValidation.Results;
+using Newtonsoft.Json;
+using PrivateHospitals.Application.Dtos.Doctor;
+using PrivateHospitals.Application.Dtos.Patient;
+using PrivateHospitals.Application.Dtos.User;
+
+public class ValidationMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    public ValidationMiddleware(RequestDelegate next)
+    {
+        _next = next;
+    }
+
+    public async Task InvokeAsync(HttpContext context)
+    {
+        if (context.Request.Method == HttpMethods.Post)
+        {
+            context.Request.EnableBuffering();
+            
+            var body = await context.Request.ReadFromJsonAsync<object>();
+
+            if (body is RegisterDoctorDto doctorDto)
+            {
+                await ValidateAndRespondAsync(context, doctorDto);
+                return;
+            }
+            else if (body is RegisterPatientDto patientDto)
+            {
+                await ValidateAndRespondAsync(context, patientDto);
+                return;
+            }
+            else if (body is LoginDto loginDto)
+            {
+                await ValidateAndRespondAsync(context, loginDto);
+                return;
+            }
```

Comment:
It's violation of SRP solid principle. Think how to this validation generic. Othervise you will have to modify it each time u add new model